cmd/rpcdaemon/commands/zkevm_api.go

```diff
@@ -1440,3 +1449,185 @@ func populateBatchDataSlimDetails(batches []*types.BatchDataSlim) (json.RawMessa
 
 	return json.Marshal(jBatches)
 }
+
+// GetProof
+func (zkapi *ZkEvmAPIImpl) GetProof(ctx context.Context, address common.Address, storageKeys []common.Hash, blockNrOrHash rpc.BlockNumberOrHash) (*accounts.SMTAccProofResult, error) {
+	api := zkapi.ethApi
+
+	tx, err := api.db.BeginRo(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback()
+	if api.historyV3(tx) {
+		return nil, fmt.Errorf("not supported by Erigon3")
+	}
+
+	blockNr, _, _, err := rpchelper.GetBlockNumber(blockNrOrHash, tx, api.filters)
+	if err != nil {
+		return nil, err
+	}
+
+	latestBlock, err := rpchelper.GetLatestBlockNumber(tx)
```

Comment:
Thanks, this is a good catch. As discussed, the best solution is to make sure `GetLatestBlockNumber` returns the block that has passed `finish` stage, because we need to make sure the block's state root hash validation has passed.